openpype/pipeline/publish/publish_plugins.py

```diff
@@ -280,3 +286,143 @@ def staging_dir(self, instance):
+        """
+        if "imageioSettings" in context.data:
+            return context.data["imageioSettings"]
+
+        project_name = context.data["projectName"]
+        host_name = context.data["hostName"]
+        anatomy_data = context.data["anatomyData"]
+        project_settings_ = context.data["project_settings"]
+
+        config_data = get_imageio_config(
+            project_name, host_name,
+            project_settings=project_settings_,
+            anatomy_data=anatomy_data
+        )
+        file_rules = get_imageio_file_rules(
+            project_name, host_name,
+            project_settings=project_settings_
+        )
+
+        # caching settings for future instance processing
+        context.data["imageioSettings"] = (config_data, file_rules)
+
+        return config_data, file_rules
+
+    def set_representation_colorspace(
+        self, representation, context,
+        colorspace=None,
+        colorspace_settings=None
+    ):
+        """Sets colorspace data to representation.
+
+        Args:
+            representation (dict): publishing representation
+            context (publish.Context): publishing context
+            config_data (dict): host resolved config data
+            file_rules (dict): host resolved file rules data
+            colorspace (str, optional): colorspace name. Defaults to None.
+            colorspace_settings (tuple[dict, dict], optional):
+                Settings for config_data and file_rules.
+                Defaults to None.
+
+        Example:
+            ```
+            {
+                # for other publish plugins and loaders
+                "colorspace": "linear",
+                "configData": {
+                    # for future references in case need
+                    "path": "/abs/path/to/config.ocio",
+                    # for other plugins within remote publish cases
+                    "template": "{project[root]}/path/to/config.ocio"
+                }
+            }
+            ```
+
+        """
+        if colorspace_settings is None:
+            colorspace_settings = self.get_colorspace_settings(context)
+
+        # unpack colorspace settings
+        config_data, file_rules = colorspace_settings
+
+        if not config_data:
+            # warn in case no colorspace path was defined
+            self.log.warning("No colorspace management was defined")
+            return
+
+        self.log.info("Config data is : `{}`".format(
+            config_data))
+
+        ext = representation["ext"]
+        # check extension
+        self.log.debug("__ ext: `{}`".format(ext))
+        if ext.lower() not in self.allowed_ext:
+            return
+
+        project_name = context.data["projectName"]
+        host_name = context.data["hostName"]
+        project_settings = context.data["project_settings"]
+
+        # get one filename
+        filename = representation["files"]
+        if isinstance(filename, list):
+            filename = filename.pop()
```

Comment:
`filename.pop()` will remove the last item from `representation["files"]` and modify the resulting representation for integration. Using `filename[0]` will fix this.

```suggestion
            filename = filename[0]
```

Reply:
Can confirm this is an error and will break publishing for sequences.